ushahidi/platform-client
# 3815 switch to eslint

This pull request makes the following changes:
- Removes jscs and jshint
- Updates eslint
- Adds new gulp tasks for eslint
- Changes pre-commit-hooks to use eslint
- Use eslint instead of jshint in test.sh

Testing checklist:
- Run `gulp eslint`
- [ ] It should run the linter (and not catch any errors if you havent made any changes to the code)
- Make a styling-error in the code
- Run `gulp eslint`
- [ ] it should find and notify about the error
- make a style-error somewhere in the code in the app-directory
- run `gulp eslintfix:app`
- [ ]   this error should be fixed
- make a style-error somewhere in the code in the test-directory
- run gulp `eslintfix:test`
- [ ]  this error should be fixed
- make a style-error somewhere in the code in both the app-directory and the test-directory
- run gulp `eslintfix`
- [ ] both errors should be fixed

- [x ] I certify that I ran my checklist

Fixes ushahidi/platform# .

Ping @ushahidi/platform
